keps/sig-scheduling/20190926-default-even-pod-spreading.md

```diff
@@ -281,6 +281,9 @@ Alpha (v1.17):
 - [ ] Predicate implementation.
 - [ ] Test cases mentioned in the [Test Plan](#test-plan).
 
+Default `topologySpreadConstaints` will graduate from Alpha along with
```

Comment:
It's a little confusing to me. Are you trying to say (1) it will become Alpha in 1.17, and (2) it won't be promoted to Beta earlier than `EvenPodsSpread` feature?

Reply:
They would be promoted to beta together.

Reply:
Then let's reword it to:

```
Default `topologySpreadConstaints` (this feature) will graduate from Alpha along with or later than `EvenPodsSpread` feature.
```

Reply:
I think we actually want to graduate them together. Having "Even Pods Spreading" graduate to beta without the Default implies that we have to maintain from then the support of `EvenPodSpreadPriority` and `SelectorSpreadingPriority` running together.
And we want to avoid that.

Reply:
We _can_ promote them together. But stating "no early than" is _technically_ correct, and give us a buffer.

Reply:
What I'm saying is that they have to graduate together so that we can remove `SelectorSpreadingPriority`. WDYT @ahg-g ?

Reply:
yes, they should graduate in lockstep, they should use the same feature flag in fact.

Reply:
@Huang-Wei do you have any concerns with using the same flag for both?

Reply:
@alculquicondor I rethink this a bit. Even if we promote them altogether to beta in some release, users can still choose to disable one and enable the other, right? In that case, it's still in conflict with `SelectorSpreadingPriority`. So in other words, `SelectorSpreadingPriority` still needs to be kept before the `EvenPodsSpread` and `DefaultTopologySpreadConstraints` both promote to GA. We can't remove `SelectorSpreadingPriority` in Beta phase.

@ahg-g Using the same flag for both seems to be a good idea. So that we don't need to consider the complex cases that `EvenPodsSpread` is enabled while `DefaultTopologySpreadConstraints` is disabled, or vice verse; we need to only consider they're both enabled or disabled. @alculquicondor to make our logic simpler, I think we should use one featuregate, WDYT?

Reply:
> @alculquicondor I rethink this a bit. Even if we promote them altogether to beta in some release, users can still choose to disable one and enable the other, right? In that case, it's still in conflict with `SelectorSpreadingPriority`. So in other words, `SelectorSpreadingPriority` still needs to be kept before the `EvenPodsSpread` and `DefaultTopologySpreadConstraints` both promote to GA. We can't remove `SelectorSpreadingPriority` in Beta phase.

Yeah, I think it is implied that if the feature gate is disabled, then  SelectorSpreadingPriority will be executed.

Reply:
Sharing the feature gate was implied, but I just added that explicitly.

For the nuances of enabling the flag or having a policy that uses `SelectorSpreadingPriority`, see https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/20190926-default-even-pod-spreading.md#implementation-detailsnotesconstraints
Let me know if you want to suggest any changes.